warehouse/query-core/src/main/java/datawave/query/transformer/ExcerptTransform.java

```diff
@@ -337,84 +347,88 @@ private Set<Excerpt> getExcerpts(PhraseIndexes phraseIndexes) {
+        // The main difference in the second attempt is that it runs with an expanded range to allow us to remove the
+        // stop words and still have a correctly sized excerpt
         for (int attempt = 0; attempt <= 1; attempt++) {
             // if this is the first attempt, set the start and end offsets using the passed in values
             if (attempt == 0) {
-                excerptIteratorOptions.put(TermFrequencyExcerptIterator.START_OFFSET, String.valueOf(start));
-                excerptIteratorOptions.put(TermFrequencyExcerptIterator.END_OFFSET, String.valueOf(end));
+                excerptIteratorOptions.put(START_OFFSET, String.valueOf(start));
+                excerptIteratorOptions.put(END_OFFSET, String.valueOf(end));
             } else {
-                // if this is the second attempt, set up the iterator with a bigger range by adding/subtracting the start and end offsets by "expandedSize"
-                if (start - expandSize > 0) {
-                    excerptIteratorOptions.put(TermFrequencyExcerptIterator.START_OFFSET, String.valueOf(start - expandSize));
-                } else {
-                    excerptIteratorOptions.put(TermFrequencyExcerptIterator.START_OFFSET, String.valueOf(0));
+                // if this is the second attempt, set up the iterator with a larger range by adding/subtracting
+                // the start and end offsets by "expandedSize"
+                int expandedStart = Math.max(start - expandSize, 0);
+                int expandedEnd = end + expandSize;
+                excerptIteratorOptions.put(START_OFFSET, String.valueOf(expandedStart));
+                excerptIteratorOptions.put(END_OFFSET, String.valueOf(expandedEnd));
+
+                if (log.isDebugEnabled()) {
+                    log.debug("size of excerpt requested: {}", excerptFields.getOffset(field) * 2);
+                    log.debug("original range is ({},{}) and the expanded range is ({},{})", start, end, expandedStart, expandedEnd);
                 }
-                excerptIteratorOptions.put(TermFrequencyExcerptIterator.END_OFFSET, String.valueOf(end + expandSize));
-            }
-            if (log.isDebugEnabled() && attempt == 1) {
-                log.debug("size of excerpt requested: {}", excerptFields.getOffset(field) * 2);
-                log.debug("original range is ({},{}) and the expanded range is ({},{})", start, end,
-                                excerptIteratorOptions.get(TermFrequencyExcerptIterator.START_OFFSET),
-                                excerptIteratorOptions.get(TermFrequencyExcerptIterator.END_OFFSET));
+
+                // if this is the second attempt, we want the iterator to trim the excerpt down to the size we want.
+                // (remember we run the iterator with an expanded range the second time, so we can potentially have a
+                // bigger excerpt than needed even after removing stop words)
+                excerptIterator.setTrimExcerpt(true);
```

Comment:
I think this has to be set to true after we initialize the iterator. If we set it before the initialization, it will be overridden with "false" during initialization